prompttools/experiment/experiments/ollama_experiment.py

```diff
@@ -0,0 +1,130 @@
+        else:
+            self.call_params["prompt"] = prompt
+
+        self.call_params["model_name"] = model_name
+
+        # Set defaults
+        for param in self.MODEL_PARAMETERS:
+            if param not in self.model_params:
+                self.model_params[param] = self.DEFAULT[param]
+        for param in self.CALL_PARAMETERS:
+            if param not in self.call_params:
+                self.call_params[param] = self.DEFAULT[param]
+        self.all_args = self.model_params | self.call_params
+        super().__init__()
+
+    @classmethod
+    def initialize(cls, test_parameters: dict[str, list], frozen_parameters: dict):
+        raise NotImplementedError(
+            "`initialize` is currently not compatible with `LlamaCppExperiment`. Please" "use `__init__` directly."
+        )
+    def prepare(self) -> None:
+        r"""
+        Creates argument combinations by taking the cartesian product of all inputs.
+        """
+        self.model_argument_combos = [
+            dict(zip(self.model_params, val)) for val in itertools.product(*self.model_params.values())
+        ]
+        self.call_argument_combos = [
+            dict(zip(self.call_params, val)) for val in itertools.product(*self.call_params.values())
+        ]
+
+   
+
+    def ollama_completion_fn(
+        self,
+        **params: Dict[str, Any],
+    ):
+        r"""
+        Local model helper function to make request
+        """
+        client = params["client"]
+        call_params = {k: v for k, v in params.items() if k != "client"}
+        print(call_params)
+        response = client(**call_params)
+        return response
+    
+
+    def run(
+            self,
+            runs: int = 1,
+        ) -> None:
+            r"""
+            Create tuples of input and output for every possible combination of arguments.
+            For each combination, it will execute `runs` times, default to 1.
+            For local models we need to run this in a single thread.
+            """
+            if not self.argument_combos:
+                logging.info("Preparing first...")
+                self.prepare()
+            self.results = []
+            for model_combo in self.model_argument_combos:
+                client = self.ollama_api.generate
+                for call_combo in self.call_argument_combos:
+                    for _ in range(runs):
+                        call_combo["client"] = client
+                        start = perf_counter()
+                        res = self.completion_fn(**call_combo)
+                        self.scores["latency"].append(perf_counter() - start)
+                        self.results.append(res)
+                        self.argument_combos.append(model_combo | call_combo)
+            if len(self.results) == 0:
+                logging.error("No results. Something went wrong.")
+                raise PromptExperimentException
+
```

Comment:
To save the results, you want to call `_construct_result_dfs(...)` here. 

See this updated [source code](https://github.com/hegelai/prompttools/blob/main/prompttools/experiment/experiments/experiment.py#L178).